### PR TITLE
EASY-1338 and EASY-1339

### PR DESCRIPTION
**Pull request by rvanheest**
fixes part of EASY-1338 and EASY-1339

#### When applied it will
* add a `files.xsd` to both `src/main/assembly/dist/bag/metadata/files/2017/09/files.xsd` and `src/main/assembly/dist/bag/metadata/files/files.xsd`
* add `gml:Polygon` to the `dcx-gml.xsd`
* add examples for `gml:Polygon` to `example1.xml`: one with only an exterior and one with both an exterior and multiple interiors (holes)
* reference the new `dcx-gml.xsd` in both `2017/09/ddm.xsd` and `ddm/ddm.xsd`
* copied the `emd` scheme files to `2017/09/` and added the polygon stuff in `2017/09/eas.xsd`. (I did not change the files `2017/09/emd.xsd`, `2017/09/qdc.xsd`, `2017/09/sdc.xsd`, `2017/09/xml.xsd`, but to be complete I copied them too.)

@DANS-KNAW/easy for review

Questions to @lindareijnhoudt:
* The changes in EMD are currently made in the new folder `2017/09/`, but these were not copied to the `md/emd` root files. I noticed this was also not done for the changes that were recently made by @PaulBoon with `archivelocation` and `archivetype`. Shouldn't these also include these latest changes?
* If you're satisfied with these changes, please merge the PR, publish these new schemes and let me know when this is done, such that I can proceed with changing `easy-ddm` and `easy-emd`.

@DANS-KNAW/easy this PR contains an addition in EMD for polygon support. See below a small excerpt of the code to be produced by the transformation from DDM to EMD.

**DDM:**
https://github.com/rvanheest-DANS-KNAW/easy-schema/blob/b071994a920e7e1ba886d3d85ee11f9c18336761/src/main/assembly/dist/docs/examples/dcx-gml/example1.xml#L78-L88

**EMD:**
```xml
<eas:spatial>
    <eas:place>A triangle between DANS, NWO and the railway station</eas:place>
    <eas:polygon eas:scheme="degrees">
        <eas:polygon-exterior>
            <eas:polygon-point>
                <eas:x>52.08110</eas:x>
                <eas:y>4.34521</eas:y>
            </eas:polygon-point>
            <eas:polygon-point>
                <eas:x>52.08071</eas:x>
                <eas:y>4.34422</eas:y>
            </eas:polygon-point>
            <eas:polygon-point>
                <eas:x>52.07913</eas:x>
                <eas:y>4.34332</eas:y>
            </eas:polygon-point>
            <eas:polygon-point>
                <eas:x>52.08110</eas:x>
                <eas:y>4.34521</eas:y>
            </eas:polygon-point>
        </eas:polygon-exterior>
    </eas:polygon>
</eas:spatial>
```

### CODE REVIEW

File: src/main/assembly/dist/bag/metadata/files/2017/09/files.xsd [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            <xs:all>
+                <xs:element name="file" type="f:dcterms-elements" maxOccurs="unbounded">
+                    <xs:annotation>
+                        <xs:documentation xml:lang="en">Container for file-specific information with qualified dcterms elements.</xs:documentation>
+                    </xs:annotation>
+                </xs:element>
+            </xs:all>
+            <xs:assert test="
+                    every $fi in f:file
+                        satisfies $fi/dcterms:format"/>
```
Comment (PaulBoon) [inline src/main/assembly/dist/bag/metadata/files/2017/09/files.xsd]: didn't know this could be done

File: src/main/assembly/dist/bag/metadata/files/2017/09/files.xsd [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                Use on dcterms:accessRights or members of its substitutionGroup.
+                dcterms:accessRights is interpreted as accessibleToRights
+            </xs:documentation>
+        </xs:annotation>
+        <xs:simpleContent>
+            <xs:restriction base="dc:SimpleLiteral">
+                <xs:simpleType>
+                    <xs:restriction base="f:EasyFileAccessCategoryType"/>
+                </xs:simpleType>
+                <xs:attribute ref="xml:lang" use="prohibited"/>
```
Comment (PaulBoon) [inline src/main/assembly/dist/bag/metadata/files/2017/09/files.xsd]: Why is this needed

File: src/main/assembly/dist/md/emd/2017/09/eas.xsd [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+            </xs:sequence>
+            <xs:attribute name="scheme" type="xs:token"/>
+            <xs:attribute name="schemeId" type="xs:string"/>
+        </xs:complexType>
+    </xs:element>
+
+    <xs:element name="polygon-exterior" type="eas:PolygonPointType"/>
+    <xs:element name="polygon-interior" type="eas:PolygonPointType"/>
+
+    <xs:complexType name="PolygonPointType">
```
Comment (PaulBoon) [inline src/main/assembly/dist/md/emd/2017/09/eas.xsd]: The name is confusing, since it is not the 'point', but the 'polygon' it is the 'type of'. 
the element "polygon-point" would be the one that might be of type 'PolygonPointType'. 
Suggestion is to change it to 'PolygonType'.

File: src/main/assembly/dist/md/emd/2017/09/eas.xsd [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+        <xs:sequence>
+            <xs:element name="place" type="xs:string" minOccurs="0"/>
```
Comment (PaulBoon) [inline src/main/assembly/dist/md/emd/2017/09/eas.xsd]: there is already a 'place' in the 'spatial type' (also note the lowercase 't' in "type").

File: src/main/assembly/dist/docs/examples/dcx-gml/example1.xml [hunk 5]
```diff
@@ -65,7 +65,51 @@
...
+                    <upperCorner>3 1</upperCorner>
+                </Envelope>
+            </boundedBy>
+        </dcx-gml:spatial>
+
+        <!-- polygon example -->
+        <dcx-gml:spatial>
+            <Polygon xmlns="http://www.opengis.net/gml">
+                <exterior>
+                    <LinearRing>
```
Comment (PaulBoon) [inline src/main/assembly/dist/docs/examples/dcx-gml/example1.xml]: I am only a bit worried that 'LinearRing' is not the only way to specify this in GML and that we have to support all the other options when converting to EMD. But we need to check this.

File: src/main/assembly/dist/bag/metadata/files/2017/09/files.xsd (hunk 1 above)
Comment (rvanheest) [reply src/main/assembly/dist/bag/metadata/files/2017/09/files.xsd]: ©️ @lindareijnhoudt (CC-BY-NC-SA) 😜

File: src/main/assembly/dist/bag/metadata/files/2017/09/files.xsd (hunk 2 above)
Comment (rvanheest) [reply src/main/assembly/dist/bag/metadata/files/2017/09/files.xsd]: I think because @lindareijnhoudt doesn't want to accept an `xml:lang` there?

File: src/main/assembly/dist/md/emd/2017/09/eas.xsd (hunk 3 above)
Comment (rvanheest) [reply src/main/assembly/dist/md/emd/2017/09/eas.xsd]: A polygon is a set of 0-1 exterior and 0-∞ interiors. So I can't really call this the `PolygonType` either. Maybe `PolygonPartType`? That might be better indeed.

File: src/main/assembly/dist/md/emd/2017/09/eas.xsd (hunk 4 above)
Comment (rvanheest) [reply src/main/assembly/dist/md/emd/2017/09/eas.xsd]: I know. However, the GML polygon defines `place` (called `description` there) on a `PolygonPartType` as well as on the whole `Polygon`.

File: src/main/assembly/dist/docs/examples/dcx-gml/example1.xml (hunk 5 above)
Comment (rvanheest) [reply src/main/assembly/dist/docs/examples/dcx-gml/example1.xml]: Currently the GML scheme says:

```xml
<complexType name="PolygonType">
   <complexContent>
      <extension base="gml:AbstractSurfaceType">
         <sequence>
            <element ref="gml:exterior" minOccurs="0"/>
            <element ref="gml:interior" minOccurs="0" maxOccurs="unbounded"/>
         </sequence>
      </extension>
   </complexContent>
</complexType>
<element name="Polygon"
         type="gml:PolygonType"
         substitutionGroup="gml:_Surface"/>
<element name="exterior" type="gml:AbstractRingPropertyType"/>

<complexType name="AbstractRingPropertyType">
   <sequence>
      <element ref="gml:LinearRing"/>
   </sequence>
</complexType>
```

So, I guess there is only one kind: `LinearRing`.

Comment (PaulBoon) [reply src/main/assembly/dist/docs/examples/dcx-gml/example1.xml]: Nice!

File: src/main/assembly/dist/md/emd/2017/09/eas.xsd (hunk 3 above)
Comment (PaulBoon) [reply src/main/assembly/dist/md/emd/2017/09/eas.xsd]: Ok

File: src/main/assembly/dist/bag/metadata/files/2017/09/files.xsd (hunk 2 above)
Comment (PaulBoon) [reply src/main/assembly/dist/bag/metadata/files/2017/09/files.xsd]: I forgot that this was a copy, sorry

Comment (rvanheest) [reply src/main/assembly/dist/bag/metadata/files/2017/09/files.xsd]: No, this is in fact new, but @lindareijnhoudt wrote it
